### DISCUSSION TITLE
Version 3.0.5 zu groß?

### DISCUSSION THREAD

**Discussion by Hotzi57** [beabel/radiowecker#70] [Fragen & Antworten / Q&A] [answered]
Als treuer Tochradiobastler und -betreiber wollte ich via OTA auf 3.0.5 updaten. Jetzt meint aber Arduino IDE, der Sketch sei zu groß, Zitat:
Der Sketch verwendet 1311917 Bytes (100%) des Programmspeicherplatzes. Das Maximum sind 1310720 Bytes.
Globale Variablen verwenden 64172 Bytes (19%) des dynamischen Speichers, 263508 Bytes für lokale Variablen verbleiben. Das Maximum sind 327680 Bytes.
Der Sketch ist zu groß; unter https://.....
Textbereich überschreitet den verfügbaren Platz auf der Platine

Compilation error: Textbereich überschreitet den verfügbaren Platz auf der Platine

Es sind jetzt auch wesentlich mehr registerkarten, als vorher vorhanden. Das liegt sicher am "Aufräumen", oder?

**Comment by beabel** [comment]
Bis max 3.0 passt es

**Reply by assembleer** [reply]
You can change this by using a custom partition. This can work as I use it now for a rework on the sketch.
**** my arduino ide memory with my custom partition ***
Sketch uses 1364697 bytes (92%) of program storage space. Maximum is 1474560 bytes.
Global variables use 64028 bytes (19%) of dynamic memory, leaving 263652 bytes for local variables. Maximum is 327680 bytes. 
I'm using the AZ-Delivery esp32devkitC v4 with 4MB) under Linux. So this description works for Linux, not Windows (but might perhaps with some path differences also works). However, arduino ide with esp32 core 2.0.x does not support custom partition via menu (esp32 core 3 does), but version 3.x does not work with radiowecker due to other versions of  libraries.

For custom partition:
- create file: ~/.arduino15/packages/esp32/hardware/esp32/2.0.17/boards local.txt
- content in this file: esp32.build.partitions=partitions.csv
- make a file partitions.csv in: ~/.arduino15/packages/esp32/hardware/esp32/2.0.17/variants/esp32/partitions.csv
- the content of this file (adapt to your needs):
# Name,     Type, SubType, Offset,   Size,      Flags
nvs,        data, nvs,     0x9000,   0x5000,
otadata,    data, ota,     0xE000,   0x2000,
app0,       app,  ota_0,   0x10000,  1472K,
app1,       app,  ota_1,   0x180000, 1472K,
coredump,   data, coredump,0x2F0000, 64K,
spiffs,     data, spiffs,  0x300000, 64K,

Be aware that esp32 flash alignment requires to align each partition start at a 0x1000 (4096 bytes) boundary.  Also be aware that spiffs (data) cannot be uploaded by arduino ide 2.x For this, you will need to create the spiffs partition from your data. to do this:
create /data under your sketch. Add the datafiles here. create the spiffs binary file with mkspiffs -c data -b 4096 -p 256 -s 65536 spiffs.bin (the used syntax assumes you're working dir is the sketch dir)
After that you can upload (no sketch upload needed, unless partitions or sketch need change) the spiffs.bin with: esptool write-flash 0x300000 spiffs.bin
These tools might not be default there.
also be aware. If in arduino ide the serial monitor tab is opened, the connection is busy and esptool reports it fails to connect. could not open the device. Just close the serial monitor and the command will work.

I plan to document this more compleet, but that will happen later in time.

**Comment by Hotzi57** [comment]
Danke, ich versuche es mit 2.0.17
Alles gut, das war es. Danke!

**Reply by beabel** [reply, marked as answer]
Esp32 Boardverwalter bis max 3.0.0
Sonst wird der Sketch zu groß.
Liegt nicht an mir, sondern an dem Boardverwalter

**Comment by Hotzi57** [comment]
Erledigt, Danke!

**Comment by uncle-benz** [comment]
hallo, stimmt   nicht so  ganz  "zu gross"   .wenn man   upgrate ueber  www macht  nimmt  es sogar die version 4.01 an

**Reply by beabel** [reply]
Hallo uncle-benz.

Es geht nicht um die Release Version, sondern um die BOARDVERWALTER Version 
Alle hier angebotenen Releases passen auf das Board.

**Comment by Hotzi57** [comment]
Das klingt nach einem Bibliothek Problem. Du nust exakt dur angegebenen
Versionen verwenden, dazu auch die exakten Boardverwaltung! Mit der neusten
Arduino IDE Verdion wird das sicher nichts.

Mit freundlichen Grüßen, A.Kreher

uncle-benz ***@***.***> schrieb am Sa., 12. Juli 2025, 19:28:
